holmes/core/tool_calling_llm.py

```
@@ -494,7 +499,63 @@ def call(  # type: ignore
 
         raise Exception(f"Too many LLM calls - exceeded max_steps: {i}/{max_steps}")
 
-    def _invoke_tool(
+    def _directly_invoke_tool(
+        self,
+        tool_name: str,
+        tool_params: dict,
+        user_approved: bool,
+        trace_span=DummySpan(),
```

Ruff (0.12.2)

Do not perform function call `DummySpan` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)